GeometricSeqPhenotype.java

```diff
@@ -0,0 +1,308 @@
+        traitA = tA;
+        traitB = tB;
+        this.nucleotideSequence = startingSequence;
+        this.eMutation = e;
+        this.neMutation = ne;
+        checkRep();
+    }
+
+    /**
+     * Return x-coordinate of where this GeometricSeqPhenotype is in Euclidean space
+     *
+     * @return position of this GeometricSeqPhenotype along the x-axis
+    */
+    public double getTraitA() {
+        return traitA;
+    }
+
+    /**
+     * Return y-coordinate of where this GeometricSeqPhenotype is in Euclidean space
+     *
+     * @return position of this GeometricSeqPhenotype along the y-axis
+     */
+    public double getTraitB() {
+        return traitB;
+    }
+
+    public String getSequence() {
+        return String.valueOf(this.nucleotideSequence);
+    }
+
+    /**
+     * Return the (Euclidean) distance between this GeometricSeqPhenotype and p in Euclidean space
+     *
+     * @param p  GeometricSeqPhenotype to calculate the distance between
+     * @return distance between this GeometricSeqPhenotype and p
+     */
+    public double distance(Phenotype p) {
+        GeometricSeqPhenotype p2d = (GeometricSeqPhenotype) p;
+        double distA = (getTraitA() - p2d.getTraitA());
+        double distB = (getTraitB() - p2d.getTraitB());
+        double dist = (distA * distA) + (distB * distB);
+        dist = Math.sqrt(dist);
+        checkRep();
+        return dist;
+    }
+
+    /**
+     * Returns a mutated copy of this GeometricSeqPhenotype (point substitution), original GeometricSeqPhenotype is unharmed
+     *
+     * @return a mutated copy of this GeometricSeqPhenotype
+     */
+    public Phenotype mutate() {
+        // Implementation:
+        // Mutates a nucleotide in the sequence at a random index
+        // If mutation creates a stop codon, then mutate at another index
+
+        // this.nucleotideSequence is represented using a char[] instead of
+        //  - String since Strings are immutable. New Strings have to be created to mutate a sequence,
+        //    which is messy and uses up extra heap space. Plus, Strings include an extra byte '\n'.
+        //  - StringBuffer because it's just a Wrapper around a char[], and has functionality that we don't need such as resizing.
+
+        String wildTypeAminoAcid = "", mutantAminoAcid = "";
+        int mutationIndexSiteSequence = -1;
+        char mutatedNucleotide = ' ', originalNucleotide = ' ';
+
+        // mutate at least once
+        // keep mutating if mutation generates a stop codon
+        while (mutantAminoAcid.equals("") || mutantAminoAcid.equals("STOP")) {
+            // choose random index to mutate in this.nucleotideSequence
+            mutationIndexSiteSequence = random(this.nucleotideSequence.length);
+
+            originalNucleotide = this.nucleotideSequence[mutationIndexSiteSequence];
+
+            char originalNucleotideToMutate = this.nucleotideSequence[mutationIndexSiteSequence];
```

Comment:
why do we need this variable and `originalNucleotide`?

Reply:
`originalNucleotide` is needed in order to test the transition/transversion ratio:

> // uncomment line below to test transition transversion ratio:
> // TestGeometricSeqPhenotype.mutations.println("" + originalNucleotide + mutatedNucleotide);

I can comment out  `originalNucleotide = ' '` if you think it'll make things clearer!

----

`originalNucleotideToMutate` is used to get a mutant nucleotide 
based on the transition/transversion ratio. 
> char originalNucleotideToMutate = this.nucleotideSequence[mutationIndexSiteSequence];
> // get mutant nucleotide (transition/transversion ratio)
> mutatedNucleotide = Biology.MutationType.MUTATION.getNucleotide(originalNucleotideToMutate);

I could have written it this way as well
> mutatedNucleotide = Biology.MutationType.MUTATION.getNucleotide(this.nucleotideSequence[mutationIndexSiteSequence]);